server/datastore/mysql/migrations/tables/20220708152154_IncreaseBatteryHealthLength.go

```diff
@@ -0,0 +1,20 @@
+package tables
+
+import (
+	"database/sql"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220708152154, Down_20220708152154)
+}
+
+func Up_20220708152154(tx *sql.Tx) error {
+	_, err := tx.Exec(
+		`ALTER TABLE host_batteries MODIFY health VARCHAR(20) NOT NULL;`,
```

Comment:
_"~64 kb ~20 chars ought to be enough for everyone"_?

Do we know where these strings are coming from?

Reply:
Strings are coming from the [osquery battery table](https://github.com/osquery/osquery/blob/master/specs/darwin/battery.table).

 It only lists three possibilities in the description: "Good", "Faith", or "Poor", which I assume is why Martin set VARCHAR(10). However, in testing Roberto found the string his machine returned was "Check Battery". 
 
 `Column("health", TEXT, "One of the following: \"Good\" describes a well-performing battery, \"Fair\" describes a functional battery with limited capacity, or \"Poor\" describes a battery that's not capable of providing power"),`

Reply:
OK. 
- Worth opening an issue in osquery, at least to amend docs and check possible values with them.
- Also, Roberto have you checked your battery? :)

Reply:
> Worth opening an issue in osquery, at least to amend docs and check possible values with them.

Agreed. @roperzh  would you mind opening that issue with osquery since you are able to reproduce the undocumented string?

Reply:
> would you mind opening that issue with osquery since you are able to reproduce the undocumented string?

👍 sounds good! will do.

@lucasmrod this is what I get running `osqueryi` locally:

```
osquery> select * from battery;
          manufacturer = DSY
      manufacture_date =
                 model = bq20z451
         serial_number = <redacted>
           cycle_count = 607
                health = Check Battery
             condition =
                 state = AC Power
              charging = 0
               charged = 1
     designed_capacity = 7336
          max_capacity = 5179
      current_capacity = 5179
     percent_remaining = 100
              amperage = -35
               voltage = 12467
   minutes_until_empty = 0
minutes_to_full_charge = 0
```

Now I'm wondering, is `20` enough? Looking at osquery https://github.com/osquery/osquery/blob/d2be385d71f401c85872f00d479df8f499164c5a/osquery/tables/system/darwin/battery.mm#L90-L93

The apple documentation for `kIOPSBatteryHealthKey` is not super descriptive https://developer.apple.com/documentation/iokit/kiopsbatteryhealthkey?language=objc

Reply:
Agree, I can see other values defined here: https://opensource.apple.com/source/IOKitUser/IOKitUser-514.16.2/ps.subproj/IOPSKeys.h.auto.html

Reply:
The longest string there is 28 characters: `Externally Indicated Failure`. Maybe set to 40 so we have some wiggle room?

Reply:
+1 to 40 or 50 to be safe

Reply:
+1 set to something like 40, and trim the string before insertion to the db.